discovery/db_discovery.go

```diff
@@ -108,9 +108,17 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int, suspe
 
 		// check if O's price is below B's max price
 		maxPrice := server.BroadcastCfg.MaxPrice()
-		price := big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit)
+		price, err := common.RatPriceInfo(info.PriceInfo)
+		if price == nil {
```

Comment:
I think this check should be after the error check. Otherwise, if an error is returned from `RatPriceInfo()`, price will be nil and this conditional will trigger first, but we won't log the error returned from `RatPriceInfo()`. If we do the error check first that ensures that if there is an error from `RatPriceInfo()` we'll log it.